src/frontends/pytorch/src/op/index_add.cpp

```diff
@@ -0,0 +1,105 @@
+    auto src_broadcasted = context.mark_node(
+        std::make_shared<v3::Broadcast>(alpha_src, src_broadcast_shape, BroadcastType::BIDIRECTIONAL));
+    auto src_shape_rank = get_shape_rank(context, src_broadcasted);
+    auto const_zero = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {0}));
+    auto src_rank = std::get<1>(src_shape_rank);
+    auto slice_start = context.mark_node(std::make_shared<v3::Broadcast>(const_zero, inp_rank));
+    auto axes = get_node_axes_range(context, src_broadcasted);
+    auto const_inf =
+        context.mark_node(v0::Constant::create(element::i32, Shape{1}, {std::numeric_limits<int32_t>::max()}));
+    auto slice_end = context.mark_node(std::make_shared<v3::Broadcast>(const_inf, src_rank));
+    auto slice_step = context.mark_node(std::make_shared<v3::Broadcast>(const_one, src_rank));
+    auto dim_1d = context.mark_node(std::make_shared<v3::Broadcast>(dim, const_one, BroadcastType::BIDIRECTIONAL));
+    slice_end =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(slice_end,
+                                                                       dim_1d,
+                                                                       const_one,
+                                                                       const_zero,
+                                                                       v12::ScatterElementsUpdate::Reduction::NONE));
+    auto new_shape = context.mark_node(std::make_shared<v8::Slice>(input, slice_start, slice_end, slice_step, axes));
+    new_shape = context.mark_node(std::make_shared<v3::ShapeOf>(new_shape, element::i32));
+    auto src_ =
+        context.mark_node(std::make_shared<v3::Broadcast>(src_broadcasted, new_shape, BroadcastType::BIDIRECTIONAL));
+    auto src_input_dtype = context.mark_node(std::make_shared<v1::ConvertLike>(src_, input));
+    // brodcast index to input rank size
+    src_rank = context.mark_node(std::make_shared<v3::ShapeOf>(new_shape, element::i32));
+    auto new_index_shape = context.mark_node(std::make_shared<v3::Broadcast>(const_one, src_rank));
+    auto const_minus_one = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {-1}));
+    new_index_shape =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(new_index_shape,
+                                                                       dim_1d,
+                                                                       const_minus_one,
+                                                                       const_zero,
+                                                                       v12::ScatterElementsUpdate::Reduction::NONE));
+    // precerve indicies location for spicifc dim
+    auto reshaped_index = context.mark_node(std::make_shared<v1::Reshape>(index, new_index_shape, false));
+    auto broadcasted_index =
+        context.mark_node(std::make_shared<v3::Broadcast>(reshaped_index, new_shape, BroadcastType::BIDIRECTIONAL));
```

Comment:
Same comment for bidirectional usage vs numpy

Reply:
it is requered in this place, because this broadcast is equivalent unsqueeze op for addition new axis, but index may have different size by target dim